Code/Source/Camera/ROS2CameraSensorComponent.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "Camera/ROS2CameraSensorComponent.h"
+#include "ROS2/ROS2Bus.h"
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Component/TransformBus.h>
+
+#include <sensor_msgs/image_encodings.hpp>
+#include <Utilities/ROS2Names.h>
+
+namespace ROS2
+{
+    namespace Internal
+    {
+        const char* kImageMessageType = "sensor_msgs::msg::Image";
+    }
+
+    ROS2CameraSensorComponent::ROS2CameraSensorComponent() {
+        PublisherConfiguration pc;
+        auto type = Internal::kImageMessageType;
+        pc.m_type = type;
+        pc.m_topic = "camera_image";
+        m_sensorConfiguration.m_frequency = 10;
+        m_sensorConfiguration.m_publishersConfigurations.insert(AZStd::make_pair(type, pc));
+    }
+
+    void ROS2CameraSensorComponent::Reflect(AZ::ReflectContext* context)
+    {
+        auto* serialize = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serialize)
+        {
+            serialize->Class<ROS2CameraSensorComponent, ROS2SensorComponent>()
+                    ->Version(1)
+                    ->Field("CameraName", &ROS2CameraSensorComponent::m_cameraName)
+                    ->Field("VerticalFieldOfViewDeg", &ROS2CameraSensorComponent::m_VerticalFieldOfViewDeg)
+                    ->Field("Width", &ROS2CameraSensorComponent::m_width)
+                    ->Field("Height", &ROS2CameraSensorComponent::m_height);
```

Comment:
`m_frameName`seems to be omitted in the serialization. Is this intended?

Reply:
Intended as it should not exist at all :)